src/executor.rs

```diff
@@ -654,6 +654,20 @@ fn parse_result(
                 value
             }
         },
+        CoreTypeConcrete::Bytes31(_) => match return_ptr {
+            Some(return_ptr) => JitValue::from_jit(return_ptr, type_id, registry),
+            None => {
+                #[cfg(target_arch = "x86_64")]
+                let value = JitValue::from_jit(return_ptr.unwrap(), type_id, registry);
```

Comment:
I just realized that this unwrap here would always fail if it wasn't unreachable.
It would always fail because the match's branch means that it's always `None`, and it's unreachable because x86 only returns values in two registers at most (128 bits, not 248).

Maybe we should change it to `unreachable!()`? (here and in the felt)

Reply:
You are right @azteca1998 !
I leave it like this 
```
        CoreTypeConcrete::Bytes31(_) => match return_ptr {
            // target_arch = "x86_64")]
            Some(return_ptr) => JitValue::from_jit(return_ptr, type_id, registry),
            None => {
                // target_arch = "aarch64"
                JitValue::Bytes31(unsafe {
                    *std::mem::transmute::<&[u64; 4], &[u8; 31]>(&ret_registers)
                })
            }
        },

```
Maybe in the feature we can refactor this function so it returns an error
What do you think?

Reply:
Nice catch!
I added unreachable!() for Bytes31 and Felt types. But I think we should modify the parse_result fn to return a Result, what do you think?

Reply:
Good catch! @azteca1998 
I added the unreachable!(), but I think I would be good to make a refactor in another PR that changes the fn parse_result so it return a Result, what do you think?

Reply:
Returning a result in the case of reaching an unreachable part of the code? I've never seen the point of doing that.
However, if it's a refactor that avoids the need to have an `unreachable!()` statement due to a change in the design (and not pushing it into a result) then I'm all for it.

Reply:
Yes, I meant the refactor that avoids unreachable!()
I will take it next!